yarpl/test/FlowableTest.cpp

```diff
@@ -279,116 +281,106 @@ TEST(FlowableTest, IgnoreElements) {
 }
 
 TEST(FlowableTest, IgnoreElementsPartial) {
-  auto collector = make_ref<CollectingSubscriber<int64_t>>(5);
+  auto subscriber = make_ref<TestSubscriber<int64_t>>(5);
   auto flowable = Flowables::range(0, 10)->ignoreElements();
-  flowable->subscribe(collector);
+  flowable->subscribe(subscriber);
 
-  EXPECT_EQ(
-      collector->values(),
-      std::vector<int64_t>({}));
-  EXPECT_EQ(collector->complete(), false);
-  collector->cancelSubscription();
+  EXPECT_EQ(subscriber->values(), std::vector<int64_t>({}));
+  EXPECT_FALSE(subscriber->isComplete());
+  EXPECT_FALSE(subscriber->isError());
 
-  flowable.reset();
-  collector.reset();
+  subscriber->cancel();
 }
 
 TEST(FlowableTest, IgnoreElementsError) {
-  auto collector = make_ref<CollectingSubscriber<int>>();
-  auto flowable = Flowables::error<int>(std::runtime_error("Failure"));
-  flowable->subscribe(collector);
-  EXPECT_EQ(collector->error(), true);
-  EXPECT_EQ(collector->errorMsg(), "Failure");
+  constexpr auto kMsg = "Failure";
+
+  auto subscriber = make_ref<TestSubscriber<int>>();
+  auto flowable = Flowables::error<int>(std::runtime_error(kMsg));
+  flowable->subscribe(subscriber);
+
+  EXPECT_TRUE(subscriber->isError());
+  subscriber->assertOnErrorMessage(kMsg);
 }
 
 TEST(FlowableTest, FlowableError) {
-  auto flowable = Flowables::error<int>(std::runtime_error("something broke!"));
-  auto collector = make_ref<CollectingSubscriber<int>>();
-  flowable->subscribe(collector);
+  constexpr auto kMsg = "something broke!";
+
+  auto flowable = Flowables::error<int>(std::runtime_error(kMsg));
+  auto subscriber = make_ref<TestSubscriber<int>>();
+  flowable->subscribe(subscriber);
 
-  EXPECT_EQ(collector->complete(), false);
-  EXPECT_EQ(collector->error(), true);
-  EXPECT_EQ(collector->errorMsg(), "something broke!");
+  EXPECT_FALSE(subscriber->isComplete());
+  EXPECT_TRUE(subscriber->isError());
+  subscriber->assertOnErrorMessage(kMsg);
 }
 
 TEST(FlowableTest, FlowableErrorPtr) {
+  constexpr auto kMsg = "something broke!";
+
   auto flowable = Flowables::error<int>(
-      std::make_exception_ptr(std::runtime_error("something broke!")));
-  auto collector = make_ref<CollectingSubscriber<int>>();
-  flowable->subscribe(collector);
+      std::make_exception_ptr(std::runtime_error(kMsg)));
+  auto subscriber = make_ref<TestSubscriber<int>>();
+  flowable->subscribe(subscriber);
 
-  EXPECT_EQ(collector->complete(), false);
-  EXPECT_EQ(collector->error(), true);
-  EXPECT_EQ(collector->errorMsg(), "something broke!");
+  EXPECT_FALSE(subscriber->isComplete());
+  EXPECT_TRUE(subscriber->isError());
+  subscriber->assertOnErrorMessage(kMsg);
```

Comment:
We should be using EXPECT.  It lets the test continue running and is more consistent with the rest of our gtests.

I can add a getErrorMsg() method.